zebra/zebra_rnh.c

```diff
@@ -963,9 +964,12 @@ static void copy_state(struct rnh *rnh, struct route_entry *re,
 	state->metric = re->metric;
 	state->vrf_id = re->vrf_id;
 	state->status = re->status;
-	state->ng = nexthop_group_new();
 
```

Comment:
Would it be better to expose `zebra_nhg_copy()` and use it here?

We would probably need to refactor the `zebra_router_get_next_sequence()` call out of that function though.

Reply:
hmm - there's quite a bit of extra nhg-specific stuff in `zebra_nhg_copy()` - I'm not sure it'd be worthwhile to really re-do that. but I think it'd work to use `nexthop_group_copy()`, which at least encaps the messing around with the internal list of nexthops. would that be better?

Reply:
Yes that would be better.

You don't think that other nhg-specific stuff might be useful in the future for comparing state in this code? That was mostly why I suggested it.

Reply:
I guess I was thinking that at the moment, the nh tracking code just understands nexthops at this point, so I was just trying to keep that going. I think I'd rather defer using the actual nhg-entry object until we understand how the nht code would use it?

Reply:
sure, sounds good to me

Reply:
thanks - I've pushed that (more limited) change; rerunning CI